text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+which is 7-10 digits in length and is assigned by a GS1 Member Organization. A 
+Location Reference number follows which is 2-5 digits in length and is 
+allocated by the company to a location or party. Lastly, a single digit Check 
+Digit is calculated and applied according to a GS1 algorithm.
+
+![GLN Data Format](https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSFo3J5QULgkCWyIH3yeddRPx1H5aQusLRRq47y1n2FqTshGxs%3Ax-raw-image%3A%2F%2F%2Ff4fc819afc288cee1948ca0d40d76c3bcd56a6d270fcfeeac80fd25784eb6071&usqp=CAU)
+
+After the 10-hex-characters that are consumed by the grid namespace prefix, the 
+location, and GS1 prefixes, there are 60 hex characters remaining in the address.
+The 13 digits of the GLN can be left padded with 45-hex-character zeroes and 
+right padded with 2-hex-character zeroes to accommodate potential future storage
+associated with the GS1 Location representation, for example:
+
+    �621dee� + �04� + �01� + �000000000000000000000000000000000000000000000� +
+    13-character �numeric string� location_id + �00� // location_id == GLN
+
+A full GS1 Location address (for example purposes) would therefore be:
+
+    �621dee0401000000000000000000000000000000000000000000000123456789012800�
+
+
+## Transaction Payload and Execution
+[transaction payload and execution]: "transactionpayloadandexecution"
+
+### Location PayLoad Transaction
+
+LocationPayload contains an action enum and the associated action payload. This 
+allows for the action payload to be dispatched to the appropriate logic.
+
+Only the defined actions are available and only one action payload should be 
+defined in the LocationPayload.
+
+    message LocationPayload {        
+
+     enum Actions {
+        UNSET_ACTION = 0;
+        LOCATION_CREATE = 1;
+        LOCATION_UPDATE = 2;
+        LOCATION_DELETE = 3;
+    
+    }
+    
+    Action action = 1;
+    
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    LocationCreateAction location_create = 3;
+    LocationUpdateAction location_update = 4;
+    LocationDeleteAction location_delete = 5;
+    }
+
+### LocationCreateAction
+
+LocationCreateAction adds a new location to state. The transaction should be 
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the create transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+Validation requirements:
+- If a location with location_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must 
+belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_location for the organization, 
+otherwise the transaction is invalid.
+- If the location_namespace is GS1, the organization must contain a GS1 Company 
```

Comment:
As above, change 'location_namespace' to 'location_type'. (following instances as well)